### PR TITLE
Implement Clone for IntoIter<A> where A: Clone

### LINKED ISSUE #178: Implement Clone for IntoIter<A: Array> where A: Clone
Implement Clone for IntoIter<A: Array> where A: Clone

This would be in line with the Iter struct that is returned when calling `into_iter()` on a `Vec`.
https://doc.rust-lang.org/std/slice/struct.Iter.html

### PR DESCRIPTION
**Pull request by L0uisc**
Implemented code to fix #178. The commented impl can be uncommented once specialization is possible. Both impls were tested and passed the three added tests.

### CODE REVIEW

File: lib.rs [hunk 1]
```diff
@@ -1504,6 +1504,24 @@ pub struct IntoIter<A: Array> {
     end: usize,
 }
 
+/*impl<A: Array + Clone> Clone for IntoIter<A>
```
Comment (mbrubeck) [inline lib.rs]: You can remove this. Since we already have a specialization for `SmallVec::from`, we won't need a separate specialized impl here.

Comment (L0uisc) [reply lib.rs]: I'm just having a doubt about this: if I just keep the below impl, won't `IntoIter<A>::clone()` only work where `A::Item: Clone`? I don't know how Rust plans to implement specialization, but won't the impl which limits `A::Item` to types which implement `Clone` then prevent `Copy` types to call the less-expensive `From` impl?

Comment (L0uisc) [reply lib.rs]: Or do you mean we just provide one impl and modify it when const generics land to work well with all possible `From` implementations?

Comment (mbrubeck) [reply lib.rs]: Right.

**Pull request merged**